src/Shared/SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests/Tests/Infrastructure/GivenAProvidersRepository.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Linq;
+using FluentAssertions;
+using NUnit.Framework;
+using SFA.DAS.ProviderPayments.Calc.Shared.Infrastructure.Data.Repositories;
+using SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests.Attributes;
+
+namespace SFA.DAS.ProviderPayments.Calc.Shared.IntegrationTests.Tests.Infrastructure
+{
+    [TestFixture, SetupProviderRepository(2)]
+    public class GivenAProvidersRepository
+    {
+        private ProviderRepository _sut;
+
+        [OneTimeSetUp]
+        public void Setup()
+        {
+            _sut = new ProviderRepository(GlobalTestContext.Instance.TransientConnectionString);
+        }
+
+        [Test]
+        public void ThenReturnsAPopulatedList()
+        {
+            var result = _sut.GetAllProviders();
+            result.Count().Should().Be(2);
+        }
+
+        [Test]
+        public void ThenTheUkprnsMatch()
+        {
+            var result = _sut.GetAllProviders().OrderBy(x=>x.Ukprn).ToList();
+            result.First().Ukprn.Should().Be(1);
+            result.Last().Ukprn.Should().Be(2);
+        }
```

Comment:
You are missing some tests.. 
should return an empty list if no records found.
should return correct IlrSubmissionDateTime 

ps: use the layout of the integration tests in payments due as a template.

Reply:
In this case I don't think we need to test for Empty Records as there is no where clause and we don't need to test the Dapper layer. 

As for IlrSubmissionDateTime, we are not using this field, but I'm happy to test a value is mapped

Reply:
I mention the empty list because it has a big impact if you return null instead of an empty list.